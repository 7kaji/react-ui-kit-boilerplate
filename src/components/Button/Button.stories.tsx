import { action } from "@storybook/addon-actions";
import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import Button from "./Button";

const stories = storiesOf("Components/Button", module);

stories.add(
  "default",
  withInfo({ inline: true })(() => (
    <Button onClick={action("onClick")}>
      Default
    </Button>
  )),
);
