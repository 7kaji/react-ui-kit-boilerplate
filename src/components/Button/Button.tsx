import * as React from "react";
import StyledButton from "./styled";
export interface ButtonProps {
  /** Callback function to be called when user clicks on button */
  onClick: () => void;
  children: React.ReactNode;
}

class Button extends React.Component<ButtonProps> {
  public render() {
    return (
      <StyledButton onClick={this.props.onClick}>
        {this.props.children}
      </StyledButton>
    );
  }
}

export default Button;
