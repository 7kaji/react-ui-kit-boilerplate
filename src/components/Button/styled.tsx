import styled from "styled-components";

const StyledButton = styled.button`
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0 1em;
  background: transparent;
  color: ${({ theme }) => theme.primaryColor};
  border: 2px solid ${({ theme }) => theme.primaryColor};
`;

export default StyledButton;
