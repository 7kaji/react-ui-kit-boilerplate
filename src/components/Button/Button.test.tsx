import * as React from "react";
import { create } from "react-test-renderer";

import Button from "./Button";

describe("Button", () => {
  it("should render correctly", () => {
    const component = create(
      <Button onClick={jest.fn} />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
