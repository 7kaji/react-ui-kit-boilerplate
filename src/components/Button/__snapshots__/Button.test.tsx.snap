// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="sc-bdVaJa fuFWB"
  onClick={[Function]}
>
  Test
</button>
`;
